homeassistant/components/zwave/node_entity.py

```diff
@@ -201,17 +196,15 @@ def state(self):
         """Return the state."""
         if ATTR_READY not in self._attributes:
             return None
-        stage = ''
-        if not self._attributes[ATTR_READY]:
-            # If node is not ready use stage as sub-status.
-            stage = self._attributes[ATTR_QUERY_STAGE]
+
         if self._attributes[ATTR_FAILED]:
-            return sub_status('Dead', stage)
+            return 'dead'
         if not self._attributes[ATTR_AWAKE]:
-            return sub_status('Sleeping', stage)
+            return 'sleeping'
         if self._attributes[ATTR_READY]:
-            return sub_status('Ready', stage)
-        return stage
+            return 'ready'
+
+        return None
 
```

Comment:
We might want to remove complex stages "Ready (Sleeping) but lets not remove the stages.

Instead of `return None` at the end return `self._attributes[ATTR_QUERY_STAGE]`

Reply:
👍

Reply:
Also: A Zwave device can be sleeping most of the time. It can be very useful to know if this is Complete + Sleeping (normal state) or SomethingElse+Sleeping which should only happen if the device didn't wake up since last restart.

We could (and should) move the "complex status" logic to the frontend, however one thing that will get lost is the history of zwave state.

Reply:
What do you think about a new state for nodes that haven't reached the complete query stage? "initializing", or something like that. That would get rid of the need to have query stage as a fallback state, and since most devices should normally reach complete eventually, it seems useful to have a completely different state IMO.

Reply:
On a large network (like mine) it takes a while until all nodes are ready and it is useful to see at what stage some of them are stuck. That said - it is possible to pull that from a specific node, so I'm OK with adding "Initializing"

Reply:
I also like the idea of building the composite string on the frontend where we want to display it. I'll try to incorporate that into the polymer PR.